cmake/cmkr.cmake

```diff
@@ -0,0 +1,112 @@
+include_guard()
+
+# Disable cmkr if no cmake.toml file is found
+if(NOT EXISTS ${CMAKE_CURRENT_LIST_DIR}/cmake.toml)
+    message(STATUS "[cmkr] Not found: ${CMAKE_CURRENT_LIST_DIR}/cmake.toml")
+    macro(cmkr)
+    endmacro()
+    return()
+endif()
+
+# Add a build-time dependency on the contents of cmake.toml to regenerate the CMakeLists.txt when modified
+configure_file(${CMAKE_CURRENT_LIST_DIR}/cmake.toml ${CMAKE_CURRENT_BINARY_DIR}/cmake.toml COPYONLY)
+
+# Helper macro to execute a process (COMMAND_ERROR_IS_FATAL ANY is 3.19 and higher)
+macro(cmkr_exec)
+    execute_process(COMMAND ${ARGV} RESULT_VARIABLE CMKR_EXEC_RESULT)
+    if(NOT CMKR_EXEC_RESULT EQUAL 0)
+        message(FATAL_ERROR "cmkr_exec(${ARGV}) failed (exit code ${CMKR_EXEC_RESULT})")
+    endif()
+endmacro()
+
+# Windows-specific hack (CMAKE_EXECUTABLE_PREFIX is not set at the moment)
+if(WIN32)
+    set(CMKR_EXECUTABLE_NAME "cmkr.exe")
+else()
+    set(CMKR_EXECUTABLE_NAME "cmkr")
+endif()
+
+# Use cached cmkr if found
+if(DEFINED CACHE{CMKR_EXECUTABLE} AND EXISTS ${CMKR_EXECUTABLE})
+    message(VERBOSE "[cmkr] Found cmkr: '${CMKR_EXECUTABLE}'")
+else()
+    if(DEFINED CACHE{CMKR_EXECUTABLE})
+        message(VERBOSE "[cmkr] '${CMKR_EXECUTABLE}' not found")
+    endif()
+    set(CMKR_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/_cmkr)
+    set(CMKR_EXECUTABLE ${CMAKE_CURRENT_BINARY_DIR}/_cmkr/bin/${CMKR_EXECUTABLE_NAME} CACHE INTERNAL "Full path to cmkr executable")
+    if(NOT EXISTS ${CMKR_EXECUTABLE})
+        message(VERBOSE "[cmkr] Bootstrapping '${CMKR_EXECUTABLE}'")
+        message(STATUS "[cmkr] Fetching cmkr...")
+        if(EXISTS ${CMKR_DIRECTORY})
+            cmkr_exec(${CMAKE_COMMAND} -E rm -rf ${CMKR_DIRECTORY})
+        endif()
+        find_package(Git QUIET REQUIRED)
+        set(CMKR_REPO "https://github.com/mrexodia/cmkr")

```

Comment:
Change this after the changes are in `master`, I needed to make some changes to `cmkr` to allow for it to bootstrap itself.